wpimath/src/main/native/include/frc/trajectory/Trajectory.h

```diff
@@ -119,6 +119,16 @@ class Trajectory {
    */
   Trajectory RelativeTo(const Pose2d& pose);
 
+  /**
+   * Concatenates another trajectory to the current trajectory. The user is
+   * responsible for making sure that the end pose of this trajectory and the
+   * start pose of the other trajectory match (if that is the desired behavior).
+   *
+   * @param other The trajectory to concatenate.
+   * @return The concatenated trajectory.
+   */
+  Trajectory operator+(const Trajectory& other) const;
```

Comment:
Should we also define `operator+=`?

Reply:
I would prefer if we kept things immutable here... I know the geometry classes have that feature but in hindsight I feel like that was a mistake.